rollup/internal/controller/watcher/batch_proposer.go

```diff
@@ -59,7 +62,7 @@ type BatchProposer struct {
 }
 
 // NewBatchProposer creates a new BatchProposer instance.
-func NewBatchProposer(ctx context.Context, cfg *config.BatchProposerConfig, minCodecVersion encoding.CodecVersion, chainCfg *params.ChainConfig, db *gorm.DB, reg prometheus.Registerer) *BatchProposer {
+func NewBatchProposer(ctx context.Context, cfg *config.BatchProposerConfig, minCodecVersion encoding.CodecVersion, chainCfg *params.ChainConfig, l2BlockDB, db *gorm.DB, reg prometheus.Registerer) *BatchProposer {
```

Comment:
I don't like this change. How about don't change this function and add a dedicated function to set the replayDB.

Reply:
refactored in commit: [818add4](https://github.com/scroll-tech/scroll/pull/1645/commits/818add4fdc1052b92b070a4704e781a62e6db924).